backend/logic/printful/sync/stockUpdate.js

```diff
@@ -0,0 +1,65 @@
+const _get = require('lodash/get')
+const { decryptConfig } = require('../../../utils/encryptedConfig')
+
+const { get } = require('../api')
+const { Product } = require('../../../models')
+
+/**
+ * Updates discontinued variants and variants that are out of stock to a collection.
+ *
+ * @param {Object} shop: Shop DB object
+ * @param {Number|String} productId: External ID of the product
+ * @param {Array<Object>} variants: List of variants that are out of stock/discontinued
+ */
+const stockUpdate = async (shop, productId, variants) => {
+  const shopConfig = decryptConfig(shop.config)
+  const shopInventory = _get(shopConfig, 'inventory')
+  if (shopInventory) {
+    const variantsStock = variants.reduce(
+      (stockData, variant) => ({
+        ...stockData,
+        [variant]: 0
+      }),
+      {}
+    )
+
+    const apiKey = shopConfig.printful
+    let existingVariants = 0
+
+    if (apiKey) {
+      const productData = await get(`/store/products/${productId}`, {
+        apiKey: apiKey
+      })
+      existingVariants =
+        _get(productData, 'result.sync_product.synced', 0) - variants.length
+      existingVariants = existingVariants < 0 ? 0 : existingVariants
+    }
+
+    const updatedStockData = {
+      productId: productId.toString(),
+      shopId: shop.id,
+      stockLeft: existingVariants * 10000,
```

Comment:
This will work but seems a bit hacky:)

In case existingVariants is not zero, as opposed to using a very large value, how about using -1 to indicate the inventory is unlimited (well, until Printful notifies us it is out of stock)?

It would require to update the places in the code where we decrement stockLeft to handle -1 as a special value. But it seems more readable and would also allow to easily query those Products that have unlimited inventory in the DB?

Reply:
@nick Jakir is trying to wrap up this PR. You are more familiar than I am on the Printful inventory management... How do you feel about my suggestion to use -1 for stockLeft? Or is there a better way in your opinion to handle this?